### PR TITLE
Getting Google AppEngine working

### PR DESCRIPTION
**Pull request by seanhagen**
Took a lot of moving about, but this should make it possible for us to deploy to Google's AppEngine.

There are still a few things to sort out:

* figure out how to provide connection string to Postgresql production database
* figure out logging so that it works in AppEngine and in self-compiled-binary ( appengine.NewContext only works inside AppEngine :cry: )

### CODE REVIEW

File: app/queries.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package app
+
+var goticFiles map[string]string
+
+func init() {
+  goticFiles = make(map[string]string)
+
+  goticFiles["db/queries.sql"] = "-- name: load-user-by-id\nSELECT id, email, full_name, password, secret from users where id = $1\n\n-- name: load-token-by-id\nSELECT * from tokens where id = $1\n\n-- name: load-token-by-company\nSELECT * from tokens where company_id = $1\n\n-- name: load-token-by-company-and-key\nSELECT * from tokens where client = $1 and secret = $2\n"
```
Comment (rydrman) [inline app/queries.go]: probably best to use a multiline string here since you are putting in line breaks anyway

File: app/authentication.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package app
+
+// Routes to handle validating user login against the auth app.
+// Currently doesn't do anything because the Auth app isn't up yet. Need to get
+// that deployed so this can be fleshed out.
+
+import (
+	// "github.com/gorilla/context"
+	//"golang.org/x/crypto/bcrypt"
+	// "github.com/unrolled/render"
```
Comment (rydrman) [inline app/authentication.go]: comments :wink:

File: app/queries.go (hunk 1 above)
Comment (seanhagen) [inline app/queries.go]: This file is auto-generated with the gotic command ( actually, a forked version of the gotic command, seanhagen/gotic )

File: app/authentication.go (hunk 2 above)
Comment (seanhagen) [inline app/authentication.go]: I wonder if that would be handled by merging in #1 first

Comment (seanhagen) [inline app/authentication.go]: This one will probably have tons of conflicts I'll have to fix after #1 is merged anyways

File: models/token.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package model
+
+import (
+	//"fmt"
+	"github.com/KloudKtrl/internal/database"
+	// "github.com/satori/go.uuid"
```
Comment (rydrman) [inline models/token.go]: here too

File: models/user.go [hunk 4]
```diff
@@ -3,39 +3,22 @@ package model
 import (
 	// "fmt"
 	"github.com/KloudKtrl/internal/database"
-	"github.com/icrowley/fake"
-	"github.com/satori/go.uuid"
+	// "github.com/icrowley/fake"
+	// "github.com/satori/go.uuid"
```
Comment (rydrman) [inline models/user.go]: ...

File: app/setup.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
+package app
+
+import (
+	// "fmt"
```
Comment (rydrman) [inline app/setup.go]: fmt

File: models/user.go (hunk 4 above)
Comment (seanhagen) [inline models/user.go]: Yeah, let's merge #1, see if that takes care of a bunch of this comment stuff. At the very least it'll be a merge conflict, so I'll have to change it then anyways.

**Comment by rydrman** [comment]
mostly just some commented packages to be removed... did you want to actually wait for those todos in the pull request before merging?

**Comment by seanhagen** [comment]
Nah, but merge #1 first -- pretty sure this PR will throw up a bunch of conflicts

**Comment by rydrman** [comment]
sounds good, merged #1 , go for it

**Pull request merged**